### PR TITLE
Update triton pin to move past bugs affecting sparse kernels

### PR DESCRIPTION
**Pull request by amjames**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109562

cc @alexsamardzic @nikitaved @pearu @cpuhrsch @bhosmer

### CODE REVIEW

**Comment by amjames** [comment]
During recent work to move the trion pin (#10804) an issue was discovered affecting some of the sparse triton kernels. [That issue](https://github.com/openai/triton/issues/2229) has been [fixed upstream](https://github.com/openai/triton/pull/2262), updating the pin to include the fixes and re-enable failing tests.

**Comment by desertfire** [comment]
cc @davidberard98  who is the oncall for internal triton sync

**Comment by davidberard98** [comment]
@amjames is there a reason not to update the pin to something even newer? I see `10f59d8ce` is from Sep 7

**Comment by amjames** [comment]
@davidberard98 That commit is the fix needed, besides that no particular reason not to update it further. 

Would it be prefered if I moved it along to the most recent commit?

**Comment by davidberard98** [comment]
@amjames do you think you can choose the newest commit from the triton repo (let's say https://github.com/openai/triton/commit/8e75e392ae5009c5311cb72b508ecdb2585d152e)? Might as well pick up new changes while we're going through the effort of checking broken tests and checking perf

File: test/test_sparse_csr.py [hunk 1]
```diff
@@ -3585,7 +3585,6 @@ def test_triton_bsr_dense_bmm_error_messages(self, device, dtype):
     @dtypesIfCUDA(torch.half, *[torch.bfloat16] if SM80OrLater else [], torch.float)
     @unittest.skipIf(IS_FBCODE and IS_REMOTE_GPU, "Test requires Triton")
     @precisionOverride({torch.float16: 1e-3})
-    @unittest.skip("Disable to unblock triton pin upgrade. Details in https://github.com/pytorch/pytorch/issues/108102")
```
Comment (shunting314) [inline test/test_sparse_csr.py:3588]: Thanks for removing this!

**Comment by lezcano** [comment]
Related https://github.com/pytorch/pytorch/pull/109601 
@peterbell10 needs a newer pin to merge his Scan PR

**Comment by amjames** [comment]
Closing this in favor of @peterbell10 's PR which will include the fixes needed for sparse.

**Pull request closed without merging**